src/main/java/net/masterthought/cucumber/json/Match.java

```diff
@@ -1,13 +1,23 @@
 package net.masterthought.cucumber.json;
 
+import com.google.gson.JsonElement;
+
 public class Match {
 
     // Start: attributes from JSON file report
-    private final String location = null;
+    private final JsonElement location = null;
     // End: attributes from JSON file report
 
     public String getLocation() {
-        return location;
+        if (location.isJsonPrimitive() && location.getAsJsonPrimitive().isString()) {
+            return location.getAsString();
+        } else {
+            StringBuilder sb = new StringBuilder();
+            sb.append(location.getAsJsonObject().get("filepath").getAsJsonObject().get("filename").getAsString());
+            sb.append(":");
+            sb.append(location.getAsJsonObject().get("lines").getAsJsonObject().get("data").getAsJsonArray().get(0).getAsString());
```

Comment:
Great, just mind that some json might not be fully valid, sometimes elements are optional and then we should handle as many exceptional cases as possible to avoid complains :)